### PR TITLE
#263 Add DM Listeners to ML Item Writer

### PR DESCRIPTION
**Pull request by mchiarelli**
#263 Add DM Listeners to ML Item Writer

### CODE REVIEW

**Comment by sastafford** [comment]
Mike - can you issue a pull request against the DEV branch?  Master branch is used for releases.

File: infrastructure/src/main/java/com/marklogic/spring/batch/item/writer/MarkLogicItemWriter.java [hunk 1]
```diff
@@ -194,5 +198,24 @@ public boolean isWriteAsync() {
     public void setWriteAsync(boolean writeAsync) {
         isWriteAsync = writeAsync;
     }
+    
+
+    private WriteBatchListener writeBatchlistener = null;
+    private WriteFailureListener writeFailureListener = null;
+    
+    public void  onBatchFailure(WriteFailureListener listener) {
```
Comment (sastafford) [inline infrastructure/src/main/java/com/marklogic/spring/batch/item/writer/MarkLogicItemWriter.java]: Change this to method name to setWriteFailureListener.

File: infrastructure/src/main/java/com/marklogic/spring/batch/item/writer/MarkLogicItemWriter.java [hunk 2]
```diff
@@ -194,5 +198,24 @@ public boolean isWriteAsync() {
...
+    	if (!this.isDataMovementSdk) {
+			throw new UnsupportedOperationException("Not using Data Movement SDK");
+		}
+    		this.writeFailureListener = listener;
+    }
+    
+    public void  onBatchSuccess(WriteBatchListener listener) {
```
Comment (sastafford) [inline infrastructure/src/main/java/com/marklogic/spring/batch/item/writer/MarkLogicItemWriter.java]: Change this method name to setWriteBatchListener

**Comment by sastafford** [comment]
The only other thing I would like to see is a unit test that would set the WriteBatchListener and WriteFailureListener.  Otherwise, it looks good.

**Comment by mchiarelli** [comment]
will merge into dev

**Pull request closed without merging**